tests/net/tcp/src/main.c

```diff
@@ -1484,23 +1484,4 @@ static void test_server_timeout_out_of_order_data(void)
 	net_tcp_put(ooo_ctx);
 }
 
-/** Test case main entry */
-void test_main(void)
-{
-	ztest_test_suite(test_tcp_fn,
-			 ztest_unit_test(test_presetup),
```

Comment:
Try to put the test_presetup in the setup function? It seems like this has to be the first one.